lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -925,6 +935,17 @@ protected boolean checkConnectivity() {
 		}
 	}
 
+	@Override
+	public void onPlatformEvent(PlatformEvent event, Object... args) {
+		switch(event) {
+		case NETWORK_UP:
+			connect();
+			break;
+		default:
```

Comment:
> Do we not move the connection to DISCONNECTED for NETWORK_DOWN?

Android does that anyway, because the transport will be closed. The question is whether or not we accelerate the transition to `suspended` - but we will probably experience quite short interruptions in connectivity, so I'm not convinced there is a good use-case for that.